src/main/java/org/mule/extension/file/internal/source/DirectoryListener.java

```diff
@@ -187,14 +191,88 @@ public void onTerminate() {}
 
   @Override
   public void poll(PollContext<InputStream, LocalFileAttributes> pollContext) {
-    PollWalker walker = new PollWalker(directoryPath, pollContext);
+
+    if (pollContext.isSourceStopping()) {
+      return;
+    }
+
+    LocalFileSystem fileSystem;
+    try {
+      fileSystem = fileSystemProvider.connect();
+    } catch (Exception e) {
+      LOGGER.error(format("Could not obtain connection while trying to poll directory '%s'. %s", directoryPath.toString(),
+                          e.getMessage()),
+                   e);
+      return;
+    }
+
     try {
-      walkFileTree(directoryPath, EnumSet.of(FOLLOW_LINKS), MAX_VALUE, walker);
+      List<Result<InputStream, LocalFileAttributes>> fileList =
+          fileSystem
+              .list(config, directoryPath.toString(), recursive, matcher,
+                    config.getTimeBetweenSizeCheckInMillis(timeBetweenSizeCheck, timeBetweenSizeCheckUnit).orElse(null));
```

Comment:
add a variable with this value with meaningful name `config.getTimeBetweenSizeCheckInMillis(timeBetweenSizeCheck, timeBetweenSizeCheckUnit).orElse(null)`